pkg/threescale/backend/manager.go

```diff
@@ -220,58 +220,85 @@ func (b *Backend) Flush() {
 		}
 
 		for _, cachedApp := range service {
+			if cachedApp.reportingErr && cachedApp.authErr {
+				// nothing to do here only continue on
+				// todo add logging
+				continue
+			}
+
+			//now that we have handled the dual error case, we know we need to
+			// read from the cache and modify the entry so we take a write lock
 			app, ok := b.cache.Get(cachedApp.cacheKey)
 			if !ok {
 				continue
 			}
+
 			app.Lock()
+			fetchedState := b.applicationFromResponse(cachedApp.authResp)
```

Comment:
Do we need to lock the app just before this? Looks like `applicationFromResponse` does not need it.

Reply:
correct, fixed